app/controllers/enterprises/SelectAnyTaxpayersThisEnterpriseIsAssociatedWithController.scala

```diff
@@ -58,8 +59,8 @@ class SelectAnyTaxpayersThisEnterpriseIsAssociatedWithController @Inject()(
     implicit request =>
 
           val preparedForm = request.userAnswers.get(SelectAnyTaxpayersThisEnterpriseIsAssociatedWithPage, id) match {
-            case None => form
-            case Some(value) => form.fill(value)
+            case Some(value) if mode == CheckMode => form.fill(value)
+            case _ => form
```

Comment:
This won't pre-pop the page if they go back in NormalMode

Reply:
The agreement is just that: if you come from the CYA page, you see it pre-populated; otherwise, it is not.